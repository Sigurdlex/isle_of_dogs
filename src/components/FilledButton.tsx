import React, { ReactNode } from 'react';
import styled from 'styled-components';

interface FilledButtonProps {
  children: ReactNode;
  handleClick: () => void;
  color: string;
}
type ContainerProps = Pick<FilledButtonProps, 'color'>;

export const FilledButton = ({ children, handleClick, color }: FilledButtonProps) => (
  <Container onClick={handleClick} color={color}>
    {children}
  </Container>
);

const Container = styled.button<ContainerProps>`
  color: white;
  font-weight: bold;
  border: ${({ color }) => `2px solid ${color}`};
  background: ${({ color }) => color};
  border-radius: 4px;
  height: 2em;
  transition: all 0.2s;
  &:hover {
    color: ${({ color }) => color};
    background: transparent;
  }
`;
