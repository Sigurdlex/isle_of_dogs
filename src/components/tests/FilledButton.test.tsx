import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import 'jest-styled-components';

import { FilledButton } from '../FilledButton';

afterEach(cleanup);

const handleClick = jest.fn();
const CHILDREN = 'CHILDREN';
const COLOR = 'white';

test('<FilledButton />', () => {
  const { container } = render(
    <FilledButton handleClick={handleClick} color={COLOR}>
      {CHILDREN}
    </FilledButton>,
  );
  const element: HTMLButtonElement = container.firstChild as HTMLButtonElement;
  expect(element.textContent).toMatch(CHILDREN);
  fireEvent.click(element);
  expect(handleClick).toHaveBeenCalledTimes(1);
  expect(element).toHaveStyleRule('border', `2px solid ${COLOR}`);
  expect(element).toHaveStyleRule('background', COLOR);
  expect(element).toMatchSnapshot();
});
