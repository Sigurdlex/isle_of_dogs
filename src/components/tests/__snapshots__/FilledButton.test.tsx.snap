// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilledButton /> 1`] = `
.c0 {
  color: white;
  font-weight: bold;
  border: 2px solid white;
  background: white;
  border-radius: 4px;
  height: 2em;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
}

.c0:hover {
  color: white;
  background: transparent;
}

<button
  class="c0"
  color="white"
>
  CHILDREN
</button>
`;
